trace/backend_collector.go

```diff
@@ -398,8 +418,14 @@ func (c *BackendCollector) flushPairCache(cutoffTime time.Time) {
 	c.pairCache.Range(func(k, v interface{}) bool {
 		e := v.(*witnessWithInfo)
 		if e.observationTime.Before(cutoffTime) {
+			// Lock the witness while it is being flushed
+			// and unlock it after it is deleted from pairCache
+			e.witnessMutex.Lock()
+
 			c.queueUpload(e)
 			c.pairCache.Delete(k)
+
+			e.witnessMutex.Unlock()
```

Comment:
Similarly here. For this case, because there is not much going on between the intended unlock and the end of the function, we can just have a `defer e.witnessMutex.Unlock()` right after the call to `Lock`.